### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by Yonashenok**
## Summary 
[@Yonashenok](https://github.com/Yonashenok) I  have completed the implementation of [OOP school library: add Person, Student and Teacher classes](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_people.md) on the OOP school library according to project [requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_people.md). 

The following tasks have been implemented: 
- [x] Create class `Person`
- [x] Create Instance vars:`@id`, `@name`, and `@age`.
- [x] Create Getters for `@id`, `@name`, and `@age`
- [x] Create Setters for `@name` and `@age`.
- [x] Private method `of_age?`
- [x] Public method `can_use_services?` 
- [x] Create class `Student`
- [x]  Class `Student` Inherits from Person. 
- [x] Method `play_hooky`
- [x] Create class Teacher 
- [x] Override `can_use_services?`  

 

In addition, I have fulfilled the following default requirements:

* Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
* Used correct [GitHub flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/github_flow.md).
* Documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

I will be  looking forward to your feedback.
Thank you!

### CODE REVIEW

**Review by TedLivist — COMMENTED** [review]
## STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @Yonashenok,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### You have submitted a project that does not have a descriptive README.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    @age = age
+
+    @parent_permission = parent_permission
+  end
+
+  def can_use_services?
+    return true if @parent_permission || of_age?
+
+    false
+  end
```
Comment (TedLivist) [inline person.rb]: - [ ] **[OPTIONAL]** We can save some code here.

  ```
  def can_use_services?
    of_age? || @parent_permission
  end
  ```

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  private
+
+  def of_age?
+    return true if @age >= 18
+
+    false
+  end
```
Comment (TedLivist) [inline person.rb]: - [ ] **[OPTIONAL]** We can save some code here.

  ```
  def of_age?
    @age >= 18
  end
  ```

File: student.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# student with there classroom and skills
+class Student < Person
```
Comment (TedLivist) [inline student.rb:3]: - [ ] For this inheritance to work you need to import the Person class into this file. Kindly do that.

  ```
  require_relative './person'
  ```

File: teacher.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# teacher with there specialization and servies
+class Teacher < Person
```
Comment (TedLivist) [inline teacher.rb:3]: - [ ] For this inheritance to work you need to import the Person class into this file. Kindly do that.

  ```
  require_relative './person'
  ```

File: student.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  def initialize(class_room)
+    super()
+    @class_room = class_room
+  end
```
Comment (TedLivist) [inline student.rb]: - [ ] The other parameters for creating a person needs to be provided when creating a student. Kindly add them.

  ```
  def initialize(classroom, age, name = 'Unknown', parent_permission: true)
    super(age, name, parent_permission: parent_permission)
    @classroom = classroom
  end
  ```

File: teacher.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+  def initialize(_specialization)
+    super()
+    @specialization = specialization
+  end
```
Comment (TedLivist) [inline teacher.rb]: - [ ] The other parameters for creating a person needs to be provided when creating a teacher. Kindly add them.

  ```
  def initialize(specialization, age, name = 'Unknown', parent_permission: true)
    super(age, name, parent_permission: parent_permission)
    @specialization = specialization
  end
  ```

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Yonashenok,

Good job so far!
✔️ Classes are in different files based on convention
✔️ Implementation of private method
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Yonashenok** [comment]
image

**Review by DammyShittu — APPROVED** [review]
Hi @Yonashenok,

## APPROVED 🟢

## Highlights
✔️ Project met all technical requirements

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**